sa/database_test.go

```diff
@@ -188,8 +199,89 @@ func TestAdjustMySQLConfig(t *testing.T) {
 			"max_statement_time": "0",
 		},
 	}
-	adjustMySQLConfig(conf)
+	err = adjustMariaDBConfig(conf)
+	test.AssertNotError(t, err, "unexpected err setting server variables")
 	test.AssertDeepEquals(t, conf.Params, map[string]string{
 		"sql_mode": "'STRICT_ALL_TABLES'",
 	})
+
+	conf = &mysql.Config{
+		Params: map[string]string{
+			"dont_worry_about_integer_vars": "0",
+		},
+	}
+	err = adjustMariaDBConfig(conf)
+	test.AssertNotError(t, err, "we don't validate whether numeric variables are in the curated system var list")
+
+	conf = &mysql.Config{
+		Params: map[string]string{
+			"dont_worry_about_fp_vars": "43.21",
+		},
+	}
+	err = adjustMariaDBConfig(conf)
+	test.AssertNotError(t, err, "we don't validate whether numeric variables are in the curated system var list")
+
+	conf = &mysql.Config{
+		Params: map[string]string{
+			"dont_worry_about_bools": "true",
+		},
+	}
+	err = adjustMariaDBConfig(conf)
+	test.AssertNotError(t, err, "we don't validate whether booleans are in the curated system var list")
```

Comment:
I believe these tests have been (appropriately!) broken by the "is this a valid system variable at all?" check

Reply:
Most of these have been deleted, good catch.